lib/core.mli

```diff
@@ -99,233 +99,230 @@ val clone_element : element -> 'props -> element
     let clone_element element props =
       clone_element_internal Imports.react element props]
 
-val use_effect : (unit -> (unit -> unit) option_undefined) -> unit
+val use_effect_always :
+  ?before_render:bool -> (unit -> (unit -> unit) option_undefined) -> unit
```

Comment:
`before_render` adds cognitive load as there's nothing like this in either React or rescript-react, and also leads to more verbose code. What are the reasons why you think `use_layout_effect*` functions should be replaced with it?

Reply:
`before_render` is very descriptive, signalling that this will trigger the effect before rendering, as opposed to after, and that this is the only effect the flag will have. `use_layout_effect` on the other hand sounds like a completely different kind of effect. What is a "layout effect"? What does it do, and how is it different from an ordinary effect? Given no further context, I think the latter has a much bigger cognitive load.

It's probably not fair to not consider other context, of course, such as existing familiarity with React. But for this iteration I'm trying to imagine how I'd design the API if I did so from scratch. Then, starting from that, try to take other considerations into account. So it would be helpful to know if your concerns are solely from the perspective of someone who already knows React, or if it's more general.

Reply:
We could also still keep `useLayoutEffect` around, or provide a compatibility layer that has all the familiar react naming. The reason I'm making this "ideal"  version of the API first is because it will be easier to decide on a strategy when we can see both perspectives clearly in front of us.

> All this is explained with details in https://reactjs.org/docs/hooks-reference.html#uselayouteffect and https://reactjs.org/docs/hooks-reference.html#timing-of-effects. And prob a 100 other places :)

Thanks. I do already know this though :) It was more of a rhetorical question, with the point being that this is very non-obvious just from the name "layout effect", and for a novice would therefore raise these kinds of unnecessary questions.